sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -45,6 +45,9 @@ case class ScalaUDF(
     udfName: Option[String] = None)
   extends Expression with ImplicitCastInputTypes with NonSQLExpression {
 
+  // the user-defined functions must be deterministic.
+  final override def deterministic: Boolean = true
```

Comment:
Oh, although we decided that this must be `true`, but should we override this explicitly? Is this for allowing further optimizations like predicate pushdown?